tests/run/i5924.scala

```diff
@@ -0,0 +1,9 @@
+trait Matchers {
```

Comment:
It'd be really good to add some [bytecode tests](https://github.com/lampepfl/dotty/blob/master/compiler/test/dotty/tools/backend/jvm/DottyBytecodeTest.scala) which actually check the behavior of MoveStatics.